eng/CodeChecks.ps1

```diff
@@ -1,9 +1,21 @@
 #requires -version 5
 
+[CmdletBinding()]
+param (
+    [Parameter(Position=0)]
+    [ValidateNotNullOrEmpty()]
+    [string] $ServiceDirectory
+)
+
 $ErrorActionPreference = 'Stop'
 Set-StrictMode -Version 1
 
-$repoRoot = Resolve-Path "$PSScriptRoot/../sdk/"
+$root = "$PSScriptRoot/../sdk"
+if ($ServiceDirectory) {
+    $root += '/' + $ServiceDirectory
```

Comment:
Join-path?

Reply:
Join-path doesn't work well. In fact, in most strings I use `[IO.Path]::Combine`. Regardless, going through a cmdlet just to concat to obvious strings is heavy-handed.

Reply:
I'm not sure if we plan to run this script on non-windows OS's but as soon as we do we will need to change this to join or combine otherwise this path will fail.

Reply:
PowerShell normalizes paths, as will .NET when this is passed to dotnet. For example, it was forward slashes here which worked on windows, and backslashes in the other script which have been running and working across platforms. .NET checks dir separate char and alt dir spearate char, or more typically normalizes paths if .NET didn't do it already (which `Resolve-Path` actually does do, so it's irrelevant here).

Reply:
`Export-API` should be runable on non-windows. It's something that developers are expected to use locally.

Reply:
It is. "/" always works. It's "\" that doesn't work as intended on linux/macOS.